src/heliaFetch.ts

```diff
@@ -235,21 +234,26 @@ export class HeliaFetch {
    */
   private async getDirectoryResponse (...[cid, options]: Parameters<UnixFS['cat']>): Promise<AsyncIterable<Uint8Array>> {
     this.log('Getting directory response:', { cid, options })
-    let rootFile: UnixFSEntry | null = null
-    for await (const file of this.fs.ls(cid, { signal: options?.signal })) {
-      if (this.rootFilePatterns.includes(file.name)) {
-        this.log(`Found root file '${file.name}': `, file)
-        rootFile = file
+    let indexFile: UnixFSStats | null = null
+
+    for (const path of this.rootFilePatterns) {
+      try {
+        indexFile = await this.fs.stat(cid, {
+          ...options,
+          path
+        })
+
         break
-      } else {
-        this.log(`Skipping ${file.type} '${file.name}' in root CID because the filename is not in rootFilePatterns: ${this.rootFilePatterns}`)
+      } catch (err: any) {
+        this.log('error loading path %c/%s', cid, path, err)
       }
```

Comment:
For this, I remember you recommended using `fs.ls` instead of `fs.stat` 

> swap fs.stat() async race for root files (index.html, etc) to fs.ls iterator.. seems much faster (and less wasteful)

from https://github.com/ipfs/helia-http-gateway/pull/53. But this seems to be an improvement upon the original from https://github.com/ipfs/helia-http-gateway/pull/53/files#diff-685aae38f06a7057b61c9ef9d72086202a6729d8e63aee9395ef750557496d3eL234 for sure.

Reply:
Yeah, it was the wrong call 🤭